java/connector-node/risingwave-sink-jdbc/src/main/java/com/risingwave/connector/JDBCSink.java

```diff
@@ -111,23 +120,30 @@ private static List<String> getPkColumnNames(
         return pkColumnNames;
     }
 
-    private PreparedStatement prepareUpsertStatement(SinkRow row) {
+    private PreparedStatement prepareInsertOrUpsertStatement(SinkRow row) {
         try {
+            var preparedStmt = config.isUpsertSink() ? upsertPreparedStmt : insertPreparedStmt;
```

Comment:
Here we implicitly introduce an assumption that the `upsertPreparedStmt` has the same number of argument as `insertPreparedStmt`. 

We'd better keep the original method and implement another method `prepareInsertStatement` specially for append only sink.